CMakeLists.txt

```diff
@@ -35,6 +35,34 @@ if(is_big_endian)
 endif()
 
 cable_configure_compiler()
+add_compile_options(
+    -Wcast-qual
+    -Wcast-align
+    -Wextra-semi
+    -Wmissing-declarations
+    -Wold-style-cast
+)
+cable_add_cxx_compiler_flag_if_supported(-Wfinal-dtor-non-final-class)
+cable_add_cxx_compiler_flag_if_supported(-Wnewline-eof)
+cable_add_cxx_compiler_flag_if_supported(-Wsuggest-destructor-override)
+cable_add_cxx_compiler_flag_if_supported(-Wunreachable-code-break)
+cable_add_cxx_compiler_flag_if_supported(-Wduplicated-cond)
+cable_add_cxx_compiler_flag_if_supported(-Wduplicate-enum)
+cable_add_cxx_compiler_flag_if_supported(-Wlogical-op)
+
+option(WEVERYTHING "Enable almost all compiler warnings" OFF)
+if(WEVERYTHING)
+    add_compile_options(-Weverything)
```

Comment:
Why do I not see these on CI?
`Build Tidy (Debug)` step outputs
```
----
Generator:  Ninja
Compiler:   Clang
Build Type: Debug
LTO:        OFF
Flags:          -Dd_m3HasTracer -Wall -Wextra -Wparentheses -Wundef -Wpointer-arith -Wstrict-aliasing=2 -Werror=implicit-function-declaration -Wno-unused-function -Wno-unused-variable -Wno-unused-parameter -Wno-missing-field-initializers -Werror=shadow
Debug flags:   -g -DDEBUG=1 -ggdb -O0
Release flags: -O3 -Wfatal-errors -fomit-frame-pointer -fno-stack-check -fno-stack-protector
----
```

are these not all of the flags?

Reply:
These are wasm3 flags. We don't add ours to theirs.

Reply:
I don't see any warning emitted in `Build Tidy`, do we not generate any currently? If we did, would CI fail?

Reply:
All warnings are fixed.
But here is older build with the warning for https://github.com/wasmx/fizzy/pull/508#discussion_r484497148: https://app.circleci.com/pipelines/github/wasmx/fizzy/4478/workflows/b12515da-c249-4903-aa9f-8d3d1d5e7cbb/jobs/31809

Reply:
I mean, if it fails CI, what's the point of making them optional?

Reply:
People can still build it locally without a struggle.

Reply:
Do you mean they can build it locally without -Weverything for compilers that we don't test?

Reply:
Yes. Clang discourages usage of `-Weverything` regularly for projects so I want to follow this rule for now. While Fizzy is actively maintained it is probably not big deal, but future versions of Clang may enable something we don't want.